react-carousel-component/src/carousel2.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentIndex: 0,
+      previousIndex: props.images.length - 1,
+      nextIndex: 1,
+      intervalID: null
+    };
+    this.clickArrows = this.clickArrows.bind(this);
+    this.clickDot = this.clickDot.bind(this);
+  }
+
+  clickArrows(isIncreasing) {
+    let { currentIndex: current, previousIndex: previous, nextIndex: next } = this.state;
+    const max = this.props.images.length;
+
+    if (isIncreasing) {
+      current = parseIndex(current + 1, max);
+      previous = parseIndex(previous + 1, max);
+      next = parseIndex(next + 1, max);
+    } else {
+      current = parseIndex(current - 1, max);
+      previous = parseIndex(previous - 1, max);
+      next = parseIndex(next - 1, max);
+    }
+
+    window.location.hash = '#carousel-img-' + current;
+
+    this.setState({
+      currentIndex: current,
+      previousIndex: previous,
+      nextIndex: next
+    });
+  }
+
+  clickDot(index) {
+    window.location.hash = '#carousel-img-' + index;
+    this.setState({
+      currentIndex: index,
+      previousIndex: parseIndex(index + 1, this.props.images.length),
+      nextIndex: parseIndex(index - 1, this.props.images.length)
+    });
+  }
+
+  componentDidMount() {
+    const intervalID = setInterval(() => this.clickArrows(true), 3000);
+    this.setState({ intervalID });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.intervalID === this.state.intervalID) {
+      clearInterval(prevState.intervalID);
+      this.setState({ intervalID: setInterval(() => this.clickArrows(true), 3000) });
+    }
```

Comment:
I wonder if this would make more sense using `setTimeout()`, since the interval is constantly being restarted.